File: boj/1942/soln1942.go
/*
// 백준
// No. 1942
// Go 1.24.8
// by "nno0obb"
*/

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"time"
)

var reader = bufio.NewReader(os.Stdin)
var writer = bufio.NewWriter(os.Stdout)

func main() {
	defer writer.Flush()

	// Input
	for range 3 {
		var start, end string
		fmt.Fscanln(reader, &start, &end)

		// Logic
		startTime, _ := time.Parse("15:04:05", start)
		endTime, _ := time.Parse("15:04:05", end)
		if startTime.After(endTime) {
			endTime = endTime.Add(24 * time.Duration(time.Hour))
		}

		ans, currTime := 0, startTime
		for currTime.Before(endTime) || currTime.Equal(endTime) {
			num, _ := strconv.Atoi(currTime.Format("150405"))
			if num%3 == 0 {
				ans++
			}
			currTime = currTime.Add(time.Second)
		}

		// Output
		fmt.Fprintln(writer, ans)
	}
}

File: problem_solving/boj/1942/soln1942.py
"""
# 백준
# No. 1942
# Python 3.11.9
# by "nno0obb"
"""

from datetime import datetime, timedelta


def main():
    # Input
    for _ in range(3):
        start, end = input().split()

        # Logic
        start = datetime.strptime(start, "%H:%M:%S")
        end = datetime.strptime(end, "%H:%M:%S")

        if start > end:
            end += timedelta(days=1)

        ans, curr = 0, start
        while curr <= end:
            num = int(curr.strftime("%H%M%S"))
            if num % 3 == 0:
                ans += 1
            curr += timedelta(seconds=1)

        # Output
        print(ans)


if __name__ == "__main__":
    main()
